### PR TITLE
Update delete device endpoint 

### PR DESCRIPTION
**Pull request by MariusBaldovin**
Updated the delete device endpoint to respond 404 instead of 500 if the report is not found.

Updated two tests for delete device endpoint to test 404 response (test_delete_report_no_report and test_delete_report_no_device).

### CODE REVIEW

File: testing/api/devices/device_2.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        },
+        {
+          "question": "Is there a second IP port on the device?",
+          "answer": "Yes"
+        },
+        {
+          "question": "Can the second IP port on your device be disabled?",
+          "answer": "Yes"
+        }
+        ],
```
Comment (jboddey) [inline testing/api/devices/device_2.json:34]: Can you fix this formatting?

**Review by jboddey — CHANGES_REQUESTED** [review]

File: testing/api/devices/device_2.json (hunk 1 above)
Comment (MariusBaldovin) [reply testing/api/devices/device_2.json]: Solved

File: testing/api/devices/device_1.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{
+  "status": "Valid",
```
Comment (jboddey) [inline testing/api/devices/device_1.json:2]: The status field should not be passed into the API. Only the backend will determine whether it is valid or not.

File: testing/api/devices/device_2.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{
+  "status": "Valid",
```
Comment (jboddey) [inline testing/api/devices/device_2.json:2]: Same with this one

**Review by jboddey — CHANGES_REQUESTED** [review]

**Review by jboddey — APPROVED** [review]

**Pull request merged**